components/clarinet-deployments/src/lib.rs

```diff
@@ -453,9 +471,19 @@ pub async fn generate_default_deployment(
             }
         };
 
-        let mut contract_location = manifest.location.get_project_root_location()?;
-        contract_location.append_path(&contract_config.path)?;
-        let source = contract_location.read_content_as_utf8()?;
+        let (contract_location, source) = match file_accessor {
```

Comment:
It was this initial idea, but the wasm FileAccessor needs to return Future while the native one wouldn't. So since they have different signature, it's not possible. 
Do you confirm @lgalabru?

I agree that this logic is quite redundant

Reply:
The local version could return a future even if it doesn't need to. I guess the idea is just to avoid that unnecessary overhead if possible?

Reply:
I like futures, but I also like the idea of avoiding futures when we can :)
We have 2 versions of `generate_default_deployment`. the version in the `clarinet-deployment` crate is the `core` one and have this `optional` FileAccessor. The LSP crate uses it, but clarinet does not and for that reason wrap this `async fn generate_default_deployment(_, _, _, Option<_>)` into a  `fn generate_default_deployment(_, _, _)` - used everywhere in clarinet.

Reply:
@obycode @lgalabru I'll mark this one as resolved unless you want to add something?